Parsers/ASimAuditEvent/Parsers/vimAuditEventCiscoMeraki.yaml

```diff
@@ -0,0 +1,260 @@
+  [
+    "vpn_connectivity_change", "VPN connectivity change","Success", "Set",
+    "purging ISAKMP-SA", "Purging ISAKMP-SA","Partial", "Delete",
+    "purged ISAKMP-SA", "Purged ISAKMP-SA","Success", "Delete",
+    "ISAKMP-SA deleted", "ISAKMP-SA deleted","Success", "Delete",
+    "IPsec-SA request", "IPsec-SA request queued","Failure", "Other",
+    "failed to get sainfo", "Failed to get sainfo","Failure", "Other",
+    "failed to pre-process ph2 packet", "Failed to pre-process ph2 packet","Failure", "Other",
+    "phase2 negotiation failed", "Phase2 negotiation failed","Failure", "Other",
+    "initiate new phase 1 negotiation", "Initiate new phase 1 negotiation","Success", "Initialize",
+    "ISAKMP-SA established", "ISAKMP-SA established","Success", "Create",
+    "initiate new phase 2 negotiation", "Initiate new phase 2 negotiation","Partial", "Initialize",
+    "IPsec-SA established", "IPsec-SA established","Success", "Create",
+    "STP role", "Spanning-tree interface role change","Success", "Set",
+    "STP BPDU", "Spanning-tree guard state change", "", "",
+    "VRRP transition", "VRRP transition","Success", "Set",
+    "port status change", "Port status change", "", ""
+  ];
+  let EventSeverityLookup=datatable(EventResult: string, EventSeverity: string)[
+      "Success", "Informational",
+      "Partial", "Informational",
+      "Failure", "Low"
+  ];
+  let parser=(disabled: bool = false, starttime: datetime=datetime(null), endtime: datetime=datetime(null), eventresult: string='*', operation_has_any: dynamic=dynamic([]), eventtype_in: dynamic=dynamic([]), srcipaddr_has_any_prefix: dynamic=dynamic([])) {
+  let allData = union isfuzzy=true
+      (
+      meraki_CL
+      | project-rename LogMessage =  Message
+      ),
+      (
+      Syslog
+      | where Computer in (_ASIM_GetSourceBySourceType('CiscoMeraki'))
+      | project-rename LogMessage =  SyslogMessage
+      );
+  let PreFilteredData = allData
+      | where not(disabled)
+          and (isnull(starttime) or TimeGenerated >= starttime)
+          and (isnull(endtime) or TimeGenerated <= endtime) 
+          and LogMessage has "events"
+          and (LogMessage has_any ("vpn_connectivity_change", "status changed", "VRRP active", "VRRP passive") or LogMessage has_cs "Site-to-site" or LogMessage has_cs "Port")
+      | extend Parser = extract_all(@"(\d+.\d+)\s([\w\-\_]+)\s([\w\-\_]+)\s([\S\s]+)$", dynamic([1, 2, 3, 4]), LogMessage)[0]
+      | extend
+      Epoch = tostring(Parser[0])
+      | extend EpochTimestamp = split(Epoch, ".")
+      | extend EventStartTime = unixtime_seconds_todatetime(tolong(EpochTimestamp[0]))
+      | extend LogType = tostring(Parser[2]),
+          Substring = tostring(Parser[3])
+      | where LogType == "events"
+      and (array_length(eventtype_in) == 0 or LogMessage has_any (eventtype_in))
+      and (eventresult == "*" or LogMessage has eventresult);
```

Comment:
This filter won't work as LogMessage won't contain these values. Same filter being used at line 167, can we remove this one?
Please test all filter once.

Reply:
Done